ui/src/shared/utils/TimeMachineContainer.ts

```diff
@@ -0,0 +1,340 @@
+// Libraries
```

Comment:
This file is defines a container class that actually stores the `TimeMachine` state used in the CEO and DE. It uses a small library called [unstated](https://github.com/jamiebuilds/unstated), which can be thought of as a very light abstraction on top of the React context API (the whole library is <200 LOC).